[WIP] Memex coding guidelines

ShishKabab opened the pull request:
View it here, rather than in the diff:
https://github.com/WorldBrain/Memex/blob/docs/memex-dev-guide/docs/Coding-Guidelines.md

First todos:
- [ ] Links to TDD examples
- [x] Integrate link to Bob Martin video in nice way
- [ ] Add action links to issue templates in Workflow section

Remember when content is OK:
- [ ] Spelling/grammar check

ShishKabab committed:
Initial dev guidelines

ShishKabab committed:
Dev guidelines update

ShishKabab committed:
Small dev guidelines corrections

ShishKabab committed:
Further work on Dev guidelines

ShishKabab committed:
Add React-specific guidelines

ShishKabab commented:
@poltak I've just added some React-specific guidelines:
https://github.com/WorldBrain/Memex/blob/docs/memex-dev-guide/docs/Coding-Guidelines.md#react-specfic-guidelines

Don't know yet about state being passed down, because that's preferable to other solutions. But, I could be wrong and you have a different situation in mind.

If you're happy with this so far, I'd say to link to it in the main docs, merge it, and keep iterating on it.

poltak committed:
Fix some minor post-proof-read issues

- mostly small typos or bad word choices

poltak committed:
Link to coding guidelines in getting started doc

poltak commented:
Cheers @ShishKabab! I did a read over this stuff again, and fixed up some minor issues I found. I also added that link from the main getting started doc.

> Don't know yet about state being passed down, because that's preferable to other solutions. But, I could be wrong and you have a different situation in mind.

We've probably discussed this before, but what's the issue here exactly? Certain states will need to be passed down to stateless components/views. I don't see a problem there, but there's probably some good examples of bad ways state can be passed around

ShishKabab commented:
> what's the issue here exactly?

I don't know, just found it in your list on Notion:
https://www.notion.so/worldbrain/Creating-UI-guidelines-4ea4053b39854e3699f9ee2e3a7842f5#04d3829969b2425985f62aa4c08c6ea6

ShishKabab commented:
And thanks for the contributions! Should really enable spellcheck in my text editor  :P

blackforestboi commented:
@ShishKabab What do we do with this one I guess if there is no new information added, lets merge it in?

blackforestboi committed:
Merge branch 'develop' into docs/memex-dev-guide

TheExGenesis committed:
Guide out of date. Example in the codebase

Warning the guide is out of date at the top, suggesting an example in the codebase instead.

poltak committed:
Merge pull request #1220 from WorldBrain/react-component-tutorial-edit

Guide out of date. Example in the codebase

iamtouchskyer commented:
📚 **WIP Review**: Memex Coding Guidelines

**Current Status**: 🚧 Work in Progress  
**Type**: Documentation Enhancement

**Observation**: This PR has been open since 2021 and contains valuable coding standards for the project.

**Current TODOs**:
⏳ TDD examples links needed  
✅ Bob Martin video integration completed  
⏳ Workflow action links pending  
⏳ Spelling/grammar review needed

**Recommendations**:

1. **Priority**: Complete the remaining TODO items or document what's blocking them
2. **Consider**: Breaking this into a minimal viable version + future enhancements
3. **Action**: If this is no longer actively maintained, consider closing or reassigning

**Impact**: Having coding guidelines would significantly help new contributors and maintain code quality.

@ShishKabab - Is this still being actively developed? If you need assistance completing the remaining items, the community might be able to help.
